src/Repository/DefaultUnleashProxyRepository.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace Unleash\Client\Repository;
+
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Symfony\Component\VarExporter\Exception\LogicException;
+use Unleash\Client\Configuration\Context;
+use Unleash\Client\Configuration\UnleashConfiguration;
+use Unleash\Client\Configuration\UnleashContext;
+use Unleash\Client\DTO\DefaultProxyFeature;
+use Unleash\Client\DTO\Feature;
+use Unleash\Client\DTO\ProxyFeature;
+
+final class DefaultUnleashProxyRepository implements ProxyRepository
+{
+    public function __construct(
+        private UnleashConfiguration $configuration,
+        private ClientInterface $httpClient,
+        private RequestFactoryInterface $requestFactory
+    ) {
+    }
+
+    public function findFeatureByContext(string $featureName, ?Context $context = null): ?ProxyFeature
+    {
+        $cacheKey = $this->getCacheKey($featureName, $context);
+
+        if ($this->configuration->getCache() !== null && $this->configuration->getCache()->has($cacheKey)) {
+            $cachedFeature = $this->configuration->getCache()->get($cacheKey);
+            if (is_array($cachedFeature)) {
+                $featureData = $this->validateResponse($cachedFeature);
+                if ($featureData !== null) {
+                    return new DefaultProxyFeature($featureData);
+                }
+            }
+        }
+
+        $context ??= new UnleashContext();
+        $featureUrl = $this->configuration->getUrl() . 'frontend/features/' . $featureName;
+        $url = $this->addQuery($featureUrl, $this->contextToQueryString($context));
+
+        $request = $this->requestFactory->createRequest('GET', $url)
+            ->withHeader('Content-Type', 'application/json')
+            ->withHeader('Accept', 'application/json');
```

Comment:
I believe we'd only need the Accept header as it's a GET request, so there's no body content. But still, having an extra header doesn't hurt, so not a blocker

Reply:
Fair enough, may as well clean that up